smdebug/tensorflow/base_hook.py

```diff
@@ -180,7 +183,8 @@ def _get_default_collections(self):
         return DEFAULT_TF_COLLECTIONS
 
     def export_collections(self):
-        assert self._prepared_tensors[self.mode]
+        if not self.tape:
```

Comment:
else ?

Reply:
there is no prepare_layers when GradientTape is used. all tensors to be saved are EagerTensors, so the value is available to be written right away.

This behavior is similar to PT and MX behavior

Reply:
else, probably assert self.prepared_collections